### ISSUE TITLE
Roundtrip Tokenization Failure

### ISSUE THREAD

**Issue by YehowshuaScaled** [huggingface/transformers#27857]
### System Info

- `transformers` version: 4.36.0.dev0
- Platform: Linux-6.2.0-37-generic-x86_64-with-glibc2.37
- Python version: 3.11.4
- Huggingface_hub version: 0.19.4
- Safetensors version: 0.4.0
- Accelerate version: 0.24.1
- Accelerate config:    not found
- PyTorch version (GPU?): 2.2.0.dev20231126+rocm5.7 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

run the following:

```python3
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained(
    "mistralai/Mistral-7B-v0.1",
    add_eos_token=False,
    add_bos_token=False)

pre_roundtrip_seq = [523, 28713, 28767]

print(f"{pre_roundtrip_seq=}")
print(f"{tokenizer.decode(pre_roundtrip_seq)=}")

post_roundtrip_seq = tokenizer.encode(tokenizer.decode(pre_roundtrip_seq))

print(f"{post_roundtrip_seq=}")
print(f"{tokenizer.decode(post_roundtrip_seq)=}")
```

```python3 bug.py
pre_roundtrip_seq=[523, 28713, 28767]
tokenizer.decode(pre_roundtrip_seq)='&lt;s&gt;'
post_roundtrip_seq=[1]
tokenizer.decode(post_roundtrip_seq)='&lt;s&gt;'
```
### Expected behavior

It's quite conceivable and sometimes desirable that a language model could spit out the literal sequence "&lt;s&gt;" which is a less than followed by s followed by greater than as shown above.

As is, tokenizers would interprets this as a start token. This is clearly incorrect behavior. To get around this behavior, I've been manually instantiating Tokenizers from `SentencePiece` and passing them a model binary.

Clearly, by doing this, I can longer take advantage of much of the Transformer library including generate or Trainer, as both expect a `Transformer.Tokenizer`.

I'm not sure what a clean way to get around this is - could involve re-doing much of the `Transformer.Tokenizer` infrastructure - unless there is some special token escape sequence included with `Transformer.Tokenizer` that I'm simply not aware of.

**Comment by YehowshuaScaled** [comment]
As an example, `'&lt;s&gt;'` when having a model generate or classify HTML. I think this is a very valid use case.
